Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabSystemComponent.cpp

```diff
@@ -144,6 +146,20 @@ namespace AzToolsFramework
             GarbageCollectTemplates(false);
         }
 
+        void PrefabSystemComponent::OnSourceFilePathNameChanged(const AZStd::string_view fromPathName, const AZStd::string_view toPathName)
+        {
+            // Convert to prefab relative path
+            AZ::IO::Path fromRelativePath = m_prefabLoader.GenerateRelativePath(fromPathName);
```

Comment:
Relative paths with assets make me nervous. What happens if I rename this file to move it between scan folders?

Example 1: I rename from MyProject\MyPrefab.prefab to Gems\MyGem\Assets\MyPrefab.prefab. The relative path will be the same, but the scan folder will change.
Example 2: I rename from MyProject\MyPrefab.prefab to Gems\MyGem\Assets\DifferentPrefabName.prefab. The relative path and scan folder have changed.

Reply:
Looks like prefab system does not identify prefabs by scan folders. If a prefab file is referencing `MyPrefab.prefab` in Source field, I suppose it currently depends on this call to determine which one to look for.

```cpp
// Inside PrefabLoader::GenerateRelativePath()...
// The asset system allows for paths to be relative to multiple root folders, using a priority system.
// This request will make the input path relative to the most appropriate, highest-priority root folder.
AzToolsFramework::AssetSystemRequestBus::BroadcastResult(
    pathFound, &AzToolsFramework::AssetSystemRequestBus::Events::GenerateRelativeSourcePath, path.Native(), 
    relativePath, rootFolder);
```

Reply:
I also noticed that the editor would crash when I tried:
- first creating a `MyPrefab` prefab at `MyProject\MyPrefab.prefab` and
- then creating another prefab at `Gems\Camera\Assets\MyPrefab.prefab`

It immediately crashes during the second creation. Inside `CreatePrefabInMemory()`, new template id and DOM seem to be invalid at these lines ([PrefabPublicHandler.cpp#L243-L244](https://github.com/o3de/o3de/blob/development/Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabPublicHandler.cpp#L243-L244)).

Reply:
Yeah, we'll need to address how we store prefab asset paths in a separate task.